src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -232,7 +232,7 @@ protected function resolveExprConstFetch(Expr\ConstFetch $node)
         if (!$isFQNConstant) {
             if (method_exists($this->context, 'getFileName')) {
                 $fileName      = $this->context->getFileName();
-                $namespaceName = $this->context->getNamespaceName();
+                $namespaceName = $this->resolveScalarMagicConstNamespace();
```

Comment:
When `$this->context` is `ReflectionFileNamespace` class which has `getName` instead of `getNamespaceName` method original code (before this PR) ends up with fatal error about non-existing method call.

Luckily there is `resolveScalarMagicConstNamespace` method (not used anywhere except this PR), that specially handles that case.